aptos-move/move-examples/tic-tac-toe/sources/TicTacToe.move

```diff
@@ -0,0 +1,687 @@
+module tic_tac_toe::ttt {
+    use std::vector;
+    use std::option::{Self, Option};
+    use std::signer;
+    use std::error;
+    use aptos_framework::account;
+    use aptos_framework::event;
+
+    #[test_only]
+    use aptos_framework::account::create_account_for_test;
+
+
+
+    //// GAME CONSTANTS
+    const GRID_SIZE: u64 = 9;
+    const WIDTH_AND_HEIGHT: u64 = 3;
+    const EMPTY_CELL: u64 = 3;
+
+    //// PLAYER CONSTANTS
+    const PLAYER_X_TYPE: u64 = 1;
+    const PLAYER_O_TYPE: u64 = 2;
+
+    //// ERROR CODES
+    /// Placing a move on a (x,y) position that is already occupied
+    const EINVALID_MOVE: u64 = 0;
+    /// An address trying to claim a player that is already taken
+    const EPLAYER_TAKEN: u64 = 1;
+    /// Trying to destroy game before game has been finished
+    const EGAME_NOT_DONE: u64 = 3;
+    /// An address can only join as one player, not two
+    const ECANNOT_JOIN_AS_TWO_PLAYERS: u64 = 4;
+    /// A (x,y) move is out of bounds of the 3x3 grid
+    const EOUT_OF_BOUNDS_MOVE: u64 = 5;
+    /// Address doesn't exist in the current Game
+    const EPLAYER_NOT_IN_GAME: u64 = 6;
+    /// An attempt to place moves even though a Game is over
+    const EGAME_HAS_ALREADY_FINISHED: u64 = 7;
+
+
+    struct GameOverEvent has drop, store {
+        is_game_over: bool,
+    }
+
+    struct Player has copy, drop, store {
+        type: u64,
+        owner: address,
+    }
+
+    struct Board has drop, store {
+        vec: vector<u64>,
+        row: u64,
+        col: u64,
+    }
+
+    struct Game has key, store {
+        board: Board,
+        player_x: Option<Player>,
+        player_o: Option<Player>,
+        is_player_x_turn: bool,
+        is_game_over: bool,
+        game_over_events: event::EventHandle<GameOverEvent>,
```

Comment:
@movekevin does this make sense to you? It doesn't feel quite right to me to store the event handle in a struct that will be destroyed after the game has finished.

Reply:
Yeah if the intention here is to allow accessing game records after games are over, this event handle needs to be stored in a permanent struct